drivers/adc-dac/ad5592r/ad5593r.c

```diff
@@ -222,5 +224,19 @@ int32_t ad5593r_init(struct ad5592r_dev *dev)
 	if (ret < 0)
 		return ret;
 
-	return ad5592r_set_channel_modes(dev);
-}
\ No newline at end of file
+	ret = ad5592r_set_channel_modes(dev);
+
+	if(init_param->int_ref) {
+		if (ret < 0)
```

Comment:
you still have this here. should be changed same as above with ret after ad5592r_set_channel_modes